featurebyte/session/spark.py

```diff
@@ -30,46 +34,75 @@ class SparkSession(BaseSparkSession):
     """
 
     _no_schema_error = OperationalError
+    _connection: Optional[HiveConnection] = PrivateAttr(None)
 
     port: int
     use_http_transport: bool
     use_ssl: bool
-    access_token: Optional[str]
     source_type: SourceType = Field(SourceType.SPARK, const=True)
+    database_credential: Optional[Union[KerberosKeytabCredential, AccessTokenCredential]]
 
     def __init__(self, **data: Any) -> None:
-        super().__init__(**data)
-
         auth = None
         scheme = None
+        access_token = None
+        kerberos_service_name = None
+
+        # perform authentication initialization prior to super().__init__()
+        # so that we can use kerberos cache to authenticate storage
+        database_credential = data.get("database_credential")
+        if database_credential:
+            if isinstance(database_credential, KerberosKeytabCredential):
+                # create a temporary keytab file to specify the KDC
+                with tempfile.NamedTemporaryFile(mode="wb", suffix=".keytab") as keytab_file:
+                    keytab_file.write(database_credential.keytab)
+                    keytab_file.flush()
+                    cmd = ["kinit", "-kt", keytab_file.name, database_credential.principal]
+                    subprocess.run(cmd, check=True)
```

Comment:
Will there be any concerns launching a subprocess if this code runs in the API server / an async io task?

Reply:
it should not be an issue (we run subprocess in async pytests as well) but we definitely should test this in a deployed service to be sure

Reply:
Tested end-to-end (i.e. register store, initialize schema and preview source tables). 
Found some bugs and fixed them.